### ISSUE TITLE
🐛 Formatter breaks precedence when removing parentheses sometimes.

### ISSUE THREAD

**Issue by lgarron** [rome/tools#3089]
### Environment information

```bash
macOS 12.5.1
rome v0.8.0-next.ff4153b
```

### What happened?

1. Format the following:
```js
const a = {};
const b = {};

const config = { useA: true };
if ("key" in (config.useA ? a : b)) {
	console.log(config);
}
```

The parentheses around `config.useA ? a : b` are removed:

```js
// ...
if ("key" in config.useA ? a : b) {
// ...
```

But this has the following meaning:

```js
if (("key" in config.useA) ? a : b) {
```

Fortunately, TypeScript will error if `config.useA` is a boolean (like in this case), although a bug will silently be introduced if it were replaced with a statement whose type is an `Object`.

### Expected result

Rome preserves semantics while formatting. In this case, it must not remove the relevant parentheses.

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by MichaReiser** [comment]
Thanks for reporting this bug. 

I'll close this issue as it is fixed in the latest version (see [playground](https://play.rome.tools/?code=if+%28%22key%22+in+%28config.useA+%3F+a+%3A+b%29%29+%7B%0A%09console.log%28config%29%3B%0A%7D&lineWidth=80&indentStyle=tab&quoteStyle=double&indentWidth=2&sourceType=module&typescript=false&jsx=false#aQBmACAAKAAiAGsAZQB5ACIAIABpAG4AIAAoAGMAbwBuAGYAaQBnAC4AdQBzAGUAQQAgAD8AIABhACAAOgAgAGIAKQApACAAewAKAAkAYwBvAG4AcwBvAGwAZQAuAGwAbwBnACgAYwBvAG4AZgBpAGcAKQA7AAoAfQA=)) and the fix will be part of our next release (most likely fixed by #2427)

You can disable formatting of that line using `// rome-ignore format: https://github.com/rome/tools/issues/3089` in the meantime.

**Comment by lgarron** [comment]
> the fix will be part of our next release (most likely fixed by #2427)

Thanks! Are there pre-built nightly or similar releases I can use in the meantime, or do I need to wait for a manual "proper" release?
